team/bundles/org.eclipse.team.ui/src/org/eclipse/team/internal/ui/history/LocalHistoryPage.java

```diff
@@ -562,18 +563,35 @@ protected void fillTableMenu(IMenuManager manager) {
 			if (compareAction != null)
 				manager.add(compareAction);
 			if (getContentsAction != null) {
-				ISelection sel = treeViewer.getSelection();
-				if (!sel.isEmpty()) {
-					if (sel instanceof IStructuredSelection) {
-						IStructuredSelection tempSelection = (IStructuredSelection) sel;
-						if (tempSelection.size() == 1) {
-							manager.add(new Separator("getContents")); //$NON-NLS-1$
-							manager.add(getContentsAction);
-						}
-					}
+				if (showGetContentsAction(treeViewer.getStructuredSelection())) {
+					manager.add(new Separator("getContents")); //$NON-NLS-1$
+					manager.add(getContentsAction);
+				}
+			}
+		}
+	}
+
+	/**
+	 * Ensures we have a single selection, and not an invalid selection
+	 * (such as getting the contents of the already-current local file revision)
+	 * @param sel selection to work against
+	 * @return whether the Get Contents action should be displayed to the user
+	 */
+	private boolean showGetContentsAction(ITreeSelection sel) {
+		if (sel.size() == 1) {
+			Object selectedElem = sel.getFirstElement();
+			// Issue #178 avoid presenting this action against the current
+			// revision which results in an undesired file truncation
+			if (selectedElem instanceof LocalFileRevision) {
+				if (((LocalFileRevision) selectedElem).isCurrentState()) {
```

Comment:
you can directly return the boolean expression instead of using another if

Reply:
this one I believe is clearer left legible as ("if current state, return false") vs having the negation inlined which can be missed with `return !(..quite a few characters later..).isCurrentState() `